sdk/framework/path.go

```diff
@@ -308,12 +308,11 @@ func (p *Path) helpCallback(b *Backend) OperationFunc {
 			return nil, errwrap.Wrapf("error executing template: {{err}}", err)
 		}
 
-		// The plugin type (e.g. "kv", "cubbyhole") is only assigned at the time
-		// the plugin is enabled (mounted). If specified in the request, the type
-		// will be used as part of the request/response names in the OAS document
-		var requestResponsePrefix string
-		if v, ok := req.Data["requestResponsePrefix"]; ok {
-			requestResponsePrefix = v.(string)
+		// If specified in the request, the mount path will be used as part of the
+		// request/response body names and operation id's in the OpenAPI document.
+		var mountPathWithPrefix string
+		if v, ok := req.Data["mount_path_with_prefix"]; ok {
```

Comment:
It is likely this parameter will get removed as part of the outcome of discussion in #18560

Reply:
> just noticed we dont have this in the [doc](https://developer.hashicorp.com/vault/api-docs/system/internal-specs-openapi), should we?

The doc you linked is for the `system/internal-specs-openapi` endpoint. The parameter here is for generic [helpCallback](https://github.com/hashicorp/vault/blob/98dae14e753d3f51db3232d1a6837876223d22f4/sdk/framework/path.go#L268), which, I think, applies to every endpoint. I'd rather keep it as an internal/undocumented parameter since it's here purely for OpenAPI generation purposes.

> It is likely this parameter will get removed as part of the outcome of discussion in https://github.com/hashicorp/vault/issues/18560

I think this parameter or some variation of it will still be needed to pass the mount information into the plugins. The only alternatives I can think of are:

- Somehow determine the plugin's mount info from within `vault/sdk` (I don't believe that's the case)
- Extract all mount-specific code out of `vault/sdk` into `logical_system.go`. I suppose this is possible but will not be easy - we will need to iterate all request/response names & references and append mount-specific info there.